geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -8,43 +8,60 @@
 
 package org.locationtech.geomesa.blob.api
 
-import com.typesafe.scalalogging.LazyLogging
+import java.io.{File, IOException}
+import java.nio.file.Files
+import java.nio.file.attribute.PosixFilePermission._
+import java.nio.file.attribute.PosixFilePermissions
+
 import org.locationtech.geomesa.accumulo.data.{AccumuloDataStore, AccumuloDataStoreFactory}
 import org.locationtech.geomesa.blob.core.AccumuloBlobStore
 import org.locationtech.geomesa.web.core.GeoMesaScalatraServlet
-import org.scalatra.{NotFound, Ok}
+import org.scalatra._
+import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}
 
 import scala.collection.JavaConversions._
+import scala.util.Try
+import scala.util.control.NonFatal
 
-class BlobstoreServlet extends GeoMesaScalatraServlet with LazyLogging {
+class BlobstoreServlet extends GeoMesaScalatraServlet with FileUploadSupport with GZipSupport {
   override def root: String = "blob"
 
-  var abs: AccumuloBlobStore = null
-
-  // TODO: Revisit configuration and persistence of configuration.
-  // https://geomesa.atlassian.net/browse/GEOMESA-958
-  // https://geomesa.atlassian.net/browse/GEOMESA-984
-  post("/ds") {
+  val maxFileSize: Int = Try(System.getProperty(BlobstoreServlet.maxFileSizeSysProp, "50").toInt).getOrElse(50)
+  val maxRequestSize: Int = Try(System.getProperty(BlobstoreServlet.maxRequestSizeSysProp, "100").toInt).getOrElse(100)
+  // caps blob size
+  configureMultipartHandling(
+    MultipartConfig(
+      maxFileSize = Some(maxFileSize * 1024 * 1024),
+      maxRequestSize = Some(maxRequestSize * 1024 * 1024)
+    )
+  )
+  error {
+    case e: SizeConstraintExceededException =>
+      logger.error("IO Exception in BlobstoreServlet, Error message: {} \n Stacktrace: {}", e.getMessage, e.getStackTrace)
+      RequestEntityTooLarge("Uploaded file too large!")
+    case e: IOException =>
+      logger.error("IO Exception in BlobstoreServlet, Error message: {} \n Stacktrace: {}", e.getMessage, e.getStackTrace)
+      halt(500, s"IO Exception on server: ${e.getMessage}")
```

Comment:
should use GeoMesaScalatrServlet.handleError - common place to handle exposing internal messages. Although that doesn't use halt...